modules/scaleform.lua

```diff
@@ -0,0 +1,208 @@
+    EndScaleformMovieMethod() -- End the scaleform movie method
+end
+
+---@param name string
+---@return nil
+---@description Request a scaleform method with no return value or parameters
+function Scaleform:Method(name)
+    if not self.handle then -- If the scaleform handle is nil
+        return error('Scaleform handle is nil') -- Error the scaleform handle is nil
+    end
+
+    BeginScaleformMovieMethod(self.handle, name) -- Begin the scaleform movie method
+    EndScaleformMovieMethod() -- End the scaleform movie method
+end
+
+---@param name string
+---@param type string
+---@return number|string | boolean
+---@description Request a scaleform method with a return value
+function Scaleform:MethodReturn(name, type)
+    if not self.handle then -- If the scaleform handle is nil
+        return error('Scaleform handle is nil') -- Error the scaleform handle is nil
+    end
+
+    BeginScaleformMovieMethod(self.handle, name) -- Begin the scaleform movie method
+    local result = EndScaleformMovieMethodReturnValue() -- End the scaleform movie method with a return value
+
+    local timeout = 0
+    while not IsScaleformMovieMethodReturnValueReady(result) do  -- While the return value is not ready
+        Wait(0) -- Wait 0
+        timeout = timeout + 1 -- Increment the timeout
+        if timeout > 1000 then -- If the timeout is greater than 1000
+            error(('Return value failed - [%s]'):format(name)) -- Error the timeout waiting for scaleform method return value
+            return false
+        end
+    end -- End the while loop
+
+    if type == "int" then -- If the type is an integer
+        return GetScaleformMovieMethodReturnValueInt(result) -- Get the return value as an integer
+    elseif type == "bool" then
+        return GetScaleformMovieMethodReturnValueBool(result) -- Get the return value as a boolean
+    else -- If the type is not an integer
+        return GetScaleformMovieMethodReturnValueString(result) -- Get the return value as a string
+    end
+end
+
+---@param isFullscreen boolean
+---@return nil
+---@description Set the scaleform to render in full screen
+function Scaleform:SetFullScreen(isFullscreen)
+    self.fullScreen = isFullscreen
+end
+
+---@param x number
+---@param y number
+---@param width number
+---@param height number
+---@return nil
+---@description Set the properties of the scaleform (Requires SetFullScreen to be false)
+function Scaleform:SetProperties(x, y, width, height)
+    if self.fullScreen then -- If the full screen is true
+        return error('Cannot set properties when full screen is enabled') -- Error cannot set properties when full screen is enabled
+    end
+    self.x = x
+    self.y = y
+    self.width = width
+    self.height = height
+end
+
+
+---@param name string
+---@param model string|number
+---@return nil
+---@description Create a render target for the scaleform - optional , only if you want to render the scaleform in 3D
+function Scaleform:RenderTarget(name, model)
```

Comment:
Should this not be merged into `Draw`?

Reply:
in which way?

Reply:
I'm not very knowledgeable on scaleforms, so I'm unaware if there's a use case where multiple render targets would be used, and in what way. Currently it's built where you setup a render target if you want to, and only that one is used, so I was thinking if it's a one-time thing where you only set up one render target, you can simply set it up in the beginning of the draw method and clean it up when done. If there's a use case for multiple render targets then it's possible to keep this method and also add it as an option to the constructor so you can create a render target immediately.